### PR TITLE
OCPBUGS-99273: Add origin e2e test for EgressIP orphan node skip

### PR DESCRIPTION
**Pull request by Ultimate-etamitlU**
## Summary

- Adds origin e2e test for the core OCPBUGS-99273 bug scenario: one node
  with missing host-cidrs must not block EgressIP assignment cluster-wide
- Uses two real nodes with two EgressIPs to prove the controller does not
  abort the entire assignment pass when an orphan node is present
- Aligns with upstream ovn-kubernetes e2e (PR #[6773](https://github.com/ovn-kubernetes/ovn-kubernetes/pull/6773) )

## Test approach

1. Creates a bare Node object labeled `k8s.ovn.org/egress-assignable`
   but with no host-cidrs annotation (simulates orphaned/stale node)
2. Allocates EgressIPs for two real egress-assignable nodes
3. Creates an EgressIP object with both IPs
4. Verifies both EgressIPs are assigned (controller did not abort)
5. Verifies each EgressIP maps to the correct valid node
6. Verifies neither EgressIP is assigned to the orphan node

## Related

- Upstream fix: https://github.com/ovn-org/ovn-kubernetes/pull/6773
- Upstream flake fix: https://github.com/ovn-org/ovn-kubernetes/pull/6863

## Summary by CodeRabbit

- **Tests**
  - Added coverage confirming that nodes without a host-cidrs annotation are skipped during EgressIP assignment.
  - Verified EgressIP status assigns addresses only to eligible worker nodes, excluding unconfigured orphan nodes.
  - Strengthened cleanup validation to handle already-removed nodes and report unexpected deletion failures.
  - Improved coverage of EgressIP node eligibility and assignment behavior.

### CODE REVIEW

**Comment by Ultimate-etamitlU** [comment]
@coderabbitai On pre-merge check : I followed the established pattern in the same file — existing tests at lines 146 and 269 log full node names and EgressIP maps via framework.Logf. CI clusters are ephemeral, so node names are non-sensitive hence keeping it that way.

**Comment by Ultimate-etamitlU** [comment]
@coderabbitai review

**Review by tssurya — CHANGES_REQUESTED** [review]
-1, doesn't need a E2E, please proceed with the unit tests that were added upstream

**Review by tssurya — CHANGES_REQUESTED** [review]
After checking the upstream PR description wasn't the problem that allocations weren't happening for other nodes?
So there are 5 nodes, one node being wrong was blocking us continuing for other nodes ? this test seems to be doing something different - all assignment logic is possible to test with unit tests right?
would be interesting for you to add why we think we need e2e for this

**Comment by Ultimate-etamitlU** [comment]
> After checking the upstream PR description wasn't the problem that allocations weren't happening for other nodes? 

Yes, so if one node amongst all nodes has issues it stalls others, the unit test covers assignment logic which includes missing, malformed, empty annotation and retry cache.

> So there are 5 nodes, one node being wrong was blocking us continuing for other nodes ? this test seems to be doing something different - all assignment logic is possible to test with unit tests right? 

Right, one bad node blocks controller for other nodes so this tests - the E2E there focuses on does assignment work with an orphan present . There were some recommendations taken to consideration from the coderabbit as well while developing that E2E.

> would be interesting for you to add why we think we need e2e for this

I had an E2E in upstream so I thought to get that to origin as well, but again looking to this with your thoughts  I learn and agree that if it's  able to cover within unit tests the E2E isn't needed."

I'd be turning this PR completely on unit test and would update on that soon.

**Comment by Ultimate-etamitlU** [comment]
> -1, doesn't need a E2E, please proceed with the unit tests that were added upstream

@tssurya , Also as it seems the origin doesn't have the coverage/mechanism for unit tests so I'm unsure if I should close this assuming the existing unit tests are sufficient in d/s merge. Please advise and thank you for the thoughts.

**Comment by tssurya** [comment]
not sure I fully follow, but skimming through your comment:

1. u/s unit tests are automatically run d/s already in ovnk
2. for upstream why was a new e2e test added? - was there a gap there that unit tests weren't enough to cover? if you added an e2e u/s then i would be hesitant to say no need for e2e w/o knowing more about that - but I'll leave that decision to the QA contact for your bug fix (why is it assigned to core-networking-bot?)

**Comment by Ultimate-etamitlU** [comment]
> 2. for upstream why was a new e2e test added? - was there a gap there that unit tests weren't enough to cover? if you added an e2e u/s then i would be hesitant to say no need for e2e w/o knowing more about that - but I'll leave that decision to the QA contact for your bug fix (why is it assigned to core-networking-bot?)

The e2e was appended knowing there is existing e2e to have assurance on fix.  There wasn't any gap I found that time  that unit test can't cover  - in fact I realize it's just a subset of what unit test already covers.  In sum,  I considered this change to be a **functional change** ( letting controller to skip the node )  and it gives assurance also satisfies the contributing guidelines on u/s and added e2e there and pulled same to origin here.

I'll work to get a QA and thoughts to this and meanwhile will appreciate if you have any more thoughts to it.   
Thank you for quick help!

File: test/extended/networking/egressip.go [hunk 1]
```diff
@@ -187,6 +188,59 @@ var _ = g.Describe("[sig-network][Feature:EgressIP][apigroup:operator.openshift.
 		os.RemoveAll(tmpDirEgressIP)
 	})
 
+	g.It("should skip nodes without host-cidrs annotation when assigning EgressIPs", func() {
+		g.By("Creating a bare node without host-cidrs annotation to simulate an orphan")
+		orphanNodeName := fmt.Sprintf("eip-orphan-%d", time.Now().UnixNano())
+		_, err := clientset.CoreV1().Nodes().Create(context.TODO(), &corev1.Node{
```
[the remark this answers is not in the document]
Comment (Ultimate-etamitlU) [reply test/extended/networking/egressip.go:194]: @coderabbitai :   I see that context.TODO() is the established pattern throughout this file and its helpers. The egressip_helpers.go alone has 22 uses, and the helper functions called in this test (findNodeEgressIPs, applyEgressIPObject, getEgressIP) all use context.TODO() internally. So changing it make the existing pattern inconsistent.

**Comment by Ultimate-etamitlU** [comment]
@coderabbitai resume

File: test/extended/networking/egressip.go [hunk 2]
```diff
@@ -187,6 +188,58 @@ var _ = g.Describe("[sig-network][Feature:EgressIP][apigroup:operator.openshift.
...
+		eip, err := getEgressIP(oc, egressIPObjectName)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(eip.Status.Items).To(o.HaveLen(2),
+			"Expected 2 EgressIP assignments despite orphan node presence")
+		for _, item := range eip.Status.Items {
+			framework.Logf("EgressIP %s assigned to node %s", item.EgressIP, item.Node)
+			o.Expect(item.Node).NotTo(o.Equal(orphanNodeName),
+				"EgressIP %s was incorrectly assigned to orphan node %s which has no host-cidrs annotation", item.EgressIP, orphanNodeName)
+			o.Expect(validNodes).To(o.HaveKey(item.Node),
+				"EgressIP %s assigned to unexpected node %s, valid nodes: %v", item.EgressIP, item.Node, validNodes)
```
[the remark this answers is not in the document]
Comment (Ultimate-etamitlU) [reply test/extended/networking/egressip.go:239]: Well, This is a valid behavior! The EgressIP controller can assign both IP's to same valid node. Enforcing one assignment per node would make this test rigid but it's likely to break easily as per my thoughts.  The goal here is to "Don't let the controller abort when orphan present"

File: test/extended/networking/egressip.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (Ultimate-etamitlU) [reply test/extended/networking/egressip.go:194]: Shall I mark it resolved then?

**Comment by tssurya** [comment]
> > 2. for upstream why was a new e2e test added? - was there a gap there that unit tests weren't enough to cover? if you added an e2e u/s then i would be hesitant to say no need for e2e w/o knowing more about that - but I'll leave that decision to the QA contact for your bug fix (why is it assigned to core-networking-bot?)
> 
> The e2e was appended knowing there is existing e2e to have assurance on fix. There wasn't any gap I found that time that unit test can't cover - in fact I realize it's just a subset of what unit test already covers. In sum, I considered this change to be a **functional change** ( letting controller to skip the node ) and it gives assurance also satisfies the contributing guidelines on u/s and added e2e there and pulled same to origin here.
> 

that's not how it works - we need to use our discretion for when to add a unit test v/s and e2e test and not waste CI resources.. upstream and downstream - please don't add duplicated tests both via unit and e2e - what's the point of doing it that way? either you consider it functional change so add a e2e OR you consider it a module change so add unit test specially for such bugs which are pure corner case scenario and can be easily tested with unit test

**Comment by rameshsahoo11** [comment]
LGTM (+1 review)
